### PR TITLE
run heroku tests on detection from title OR schedule

### PR DESCRIPTION
**Pull request by madhur-tandon**

### CODE REVIEW

**Review by aguschin — COMMENTED** [review]
Good work! Trying to understand how this should work. Will it require commit hash when one tries to run the tests manually? Also, can I test it out in this branch before merging to `main`?

UPD: I just did a commit assuming it will ask me for input (commit hash), but looks like this works differently.

**Comment by aguschin** [comment]
@madhur-tandon could use your help here to understand this ^ :)

File: .github/workflows/check-heroku-tests.yml
Comment (mike0sv) [inline .github/workflows/check-heroku-tests.yml:10]: We have the same schedule in current wf. Can we not duplicate it and just detect if WF was triggered by schedule? Also we talked about having some specific string in commit message to trigger heroku tests even in not-scheduled runs

File: .github/workflows/check-heroku-tests.yml
Comment (aguschin) [inline .github/workflows/check-heroku-tests.yml:78]: Since we're creating a separate workflow here, maybe we can also separate the tests that don't require all these secrets? Otherwise when an external contributor pushes the PR, we have to confirm each time so it can be run. This delays the execution. If that does look simple enough, could we do it here? Otherwise we should do it separate. This is the issue https://github.com/iterative/mlem/issues/330

Comment (madhur-tandon) [reply .github/workflows/check-heroku-tests.yml]: Using the same workflow now

File: .github/workflows/check-heroku-tests.yml
Comment (madhur-tandon) [reply .github/workflows/check-heroku-tests.yml]: Doing a check on title of the PR now

**Comment by madhur-tandon** [comment]
> @madhur-tandon could use your help here to understand this ^ :)

Earlier, I was using something known as a `workflow dispatch` -- i.e. you need to trigger an action manually.
But now, I have switched to detection of the keyword `heroku` in title of the PR

**Comment by madhur-tandon** [comment]
The PRs https://github.com/iterative/mlem/pull/457 and https://github.com/iterative/mlem/pull/458 confirm that this works 😄

**Comment by aguschin** [comment]
Looking good! Thanks @madhur-tandon!

I remember we had a discussion about doing the same for k8s/sagemaker. Don't remember the resolution. Should we do this for them also?

**Review by aguschin — APPROVED** [review]

File: .github/workflows/check-test-release.yml
Comment (aguschin) [inline .github/workflows/check-test-release.yml:124]: Ah, you didn't delete `HEROKU_API_KEY` and `HEROKU_TEAM` for the step running `pytest -k 'not heroku'`. I guess you should.

Comment (madhur-tandon) [reply .github/workflows/check-test-release.yml:122]: done, but don't merge it yet, will retest those 2 dummy PRs I opened

File: .github/workflows/check-test-release.yml
Comment (aguschin) [inline .github/workflows/check-test-release.yml:116]: Will this work if it's `Heroku`, but not `heroku`?
```suggestion
      if: ${{ github.event_name == 'schedule' || contains(github.event.pull_request.title, 'heroku') }}
```

Comment (madhur-tandon) [reply .github/workflows/check-test-release.yml:116]: `This function is not case sensitive.` -- from https://docs.github.com/en/actions/learn-github-actions/expressions#contains

File: .github/workflows/check-heroku-tests.yml
Comment (madhur-tandon) [reply .github/workflows/check-heroku-tests.yml]: schedule or title check -- should work on both now

**Comment by madhur-tandon** [comment]
> I remember we had a discussion about doing the same for k8s/sagemaker. Don't remember the resolution. Should we do this for them also?

K8s test are self contained, won't be necessary.
@mike0sv can comment on Sagemaker maybe

**Comment by aguschin** [comment]
> K8s test are self contained, won't be necessary.

how much time do they take? Speeding up CI would be a good thing also.

**Comment by madhur-tandon** [comment]
> how much time do they take? Speeding up CI would be a good thing also.

The point of this PR was not to speed up CI, it was to control flaky behaviour of heroku so we don't get unnecessary failures.

**Pull request merged**
